lib/cn/route.c

```diff
@@ -328,8 +319,7 @@ route_map_insert(
             link = &(*link)->rb_right;
         } else {
             log_err("dup route detected");
-            route_node_free(map, node);
-            return NULL;
+            return merr(EEXIST);
```

Comment:
Somewhere along the line this seems to have gotten perverted.  The original RB_INSERT() API returns NULL for successful insert and a ptr to the dup node if there's a duplicate.

Reply:
I did that without knowing that it follows the semantics of the RB_INSERT() API, sorry. I've reverted it now.